.github/workflows/pr-build.yml

```diff
@@ -78,3 +78,20 @@ jobs:
         env:
           SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
           GRADLE_OPTS: -Dorg.gradle.workers.max=1
+
+  validate-avros:
+    runs-on: ubuntu-20.04
+    steps:
+      # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
+
+      - name: validate avros
+          uses: hypertrace/github-actions/gradle@main
+          with:
+            args: avroCompatibilityCheck
```

Comment:
Does this have deps on the `build` step? If, yes, I will have to add the cache step.

Reply:
I can't recall, but you should add the cache step regardless as it caches the gradle binary too. We should probably change the avro plugin so validation does not automatically run on `check` (the reason for the -x above), but that'd be a bit of a breaking change and could cause repos to stop validating entirely, so this is more reasonable right now